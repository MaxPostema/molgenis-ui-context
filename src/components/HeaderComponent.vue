<template>
  <header>
      <header-image v-if="molgenisMenu.topLogo" :imgSrc="molgenisMenu.topLogo" :imgMaxHeight="molgenisMenu.topLogoMaxHeight"/>
      <cookie-wall cookieName="permissionforcookies" />
      <nav-bar :molgenis-menu="molgenisMenu" />
  </header>
</template>

<script>
import { MolgenisMenu } from '../types'
import HeaderImage from './HeaderImage.vue'
import NavBar from './NavBar.vue'
import CookieWall from './CookieWall.vue'

export default {
  name: 'HeaderComponent',
  components: { NavBar, CookieWall, HeaderImage },
  props: {
    molgenisMenu: MolgenisMenu
  },
  data: function () {
    return {}
  }
}
</script>
