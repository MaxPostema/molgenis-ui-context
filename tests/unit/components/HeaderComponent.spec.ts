import { shallowMount } from '@vue/test-utils'
import HeaderComponent from '../../../src/components/HeaderComponent.vue'
import molgenisMenu from '../../../src/exampleMenu'

describe('HeaderComponent.vue', () => {
  it('should match the snapshot', () => {
    const wrapper = shallowMount(HeaderComponent, {
      propsData: { molgenisMenu }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should set the padder height after children have been mounted', (done) => {
    const wrapper = shallowMount(HeaderComponent, {
      propsData: { molgenisMenu }
    })

    wrapper.vm.$nextTick(() => {
      expect(wrapper.find('.mg-sticky-header-padder').html()).toContain('height: 0px;')
      done()
    })
  })
})
