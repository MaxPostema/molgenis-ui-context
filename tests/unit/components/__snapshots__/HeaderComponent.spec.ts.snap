// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent.vue should match the snapshot 1`] = `
<header>
  <div class="mg-sticky-header-padder"></div>
  <div class="fixed-top">
    <header-image-stub imgsrc="https://www.molgenis.org/assets/img/logo_green.png" imgmaxheight="150"></header-image-stub>
    <nav-bar-stub molgenismenu="[object Object]"></nav-bar-stub>
  </div>
</header>
`;
